bleak/exc.py

```diff
@@ -8,6 +8,23 @@ class BleakError(Exception):
     pass
 
 
+class BleakDeviceNotFoundError(BleakError):
+    """
+    Exception which is raised if a device can not be found by ``connect``, ``pair`` and ``unpair``.
+    This is the case if the OS Bluetooth stack has never seen this device or it was removed and forgotten.
+    """
+
+    identifier: str
+
+    def __init__(self, identifier: str, *args: object) -> None:
+        """
+        Args:
+            identifier (str): The identifier which was used to search the device (e. g. Bluetooth address or UUID)
```

Comment:
I only added the identifier here for now. But it can also be the ``BLEDevice`` to share some more information about the device if they are available. I'm a bit ambivalent about this.

Reply:
I think it would be best to use the `BLEDevice` object to wrap the platform-specific details.

Reply:
I changed it to the BLEDevice.